SimulationRuntime/c/util/rtclock.c

```diff
@@ -402,6 +402,23 @@ double rt_ext_tp_tock(rtclock_t* tick_tp) {
   return d - min_time;
 }
 
+void rt_ext_tp_tick_realtime(rtclock_t* tick_tp) {
+  *tick_tp = mach_absolute_time();
+}
+
+double rt_ext_tp_tock_realtime(rtclock_t* tick_tp) {
+  double d=0;
+  rtclock_t tock_tp = mach_absolute_time();
+  d = (double)(tock_tp - *tick_tp);
+  return d;
+}
+
+int64_t rt_ext_tp_sync_nanosec(rtclock_t* tick_tp, uint64_t nsec)
```

Comment:
This function needs to sleep in some way in order to synchronize the clock to tick_tp + nsec number of nanoseconds. Else, I guess throw an error saying OSX cannot do this.